### DISCUSSION TITLE
Permissions issues - Cannot Run Input files create by ATOM

### DISCUSSION THREAD

**Discussion by hyeom-wisc** [idaholab/moose#16315] [Q&A Getting Started] [answered]
Hello,
I am a new user of MOOSE and have tried tutorial#01 and step #2. I have installed ATOM and created the input file and tried to run the file as shown below. However, it showed error message. It was fine when I run the existing file in the tutorial. The issue was shown whenever I recreate input files using ATOM. I don't know what causes the issue. I would really appreciate if you provide some inputs. Thanks. Sung 

(moose) hyeom@ERB914-7920:~/projects/babbler/problems$ ../babbler-opt -i pressure_diffusion.i

*** ERROR ***
Unable to open file "/home/hyeom/projects/babbler/problems/pressure_diffusion.i". Check to make sure that it exists and that you have read permission.

Stack frames: 10
0: libMesh::print_trace(std::ostream&)
1: moose::internal::mooseErrorRaw(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)
2: void mooseError<char const*>(char const*&&)
3: MooseUtils::checkFileReadable(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool, bool)
4: Parser::parse(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
5: MooseApp::setupOptions()
6: MooseApp::run()
7: ../babbler-opt(+0x41d1) [0x7f397a60c1d1]
8: __libc_start_main
9: ../babbler-opt(+0x44af) [0x7f397a60c4af]
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0
[unset]: write_line error; fd=-1 buf=:cmd=abort exitcode=1
:
system msg for write_line failure : Bad file descriptor

**Comment by GiudGiud** [comment, marked as answer]
Hi Sung,

That is very odd, there should be no issue with atom. 
Can you please run 'ls -l' in '~/projects/babbler/problems' so we can check the file permissions?

Best,
Guillaume

**Reply by hyeom-wisc** [reply]
Thanks Guillaume. Please see below.

(moose) hyeom@ERB914-7920:~/projects/babbler/problems$ ls -l
total 4
---------- 1 hyeom hyeom 1544 Nov 27 12:27 pressure_diffusion.i

In addition, when I typed "git status", it says "fatal: .git/index: index file open failed: Permission denied." I am not sure this is related to the input file issue.

Sincerely,
Sung

**Reply by GiudGiud** [reply]
The --------- are the permissions for the file (for the owner, the group of users and all users). You currently do not have read (would show an r) or write permissions (would show a w).

Can you try changing the permissions of the file? On linux it's 'chmod +rw file_or_folder' to add read and write permissions. You can check the result with 'ls -l'. You may have to sudo for this command. 

Is this run on a computing cluster? You may have to contact your platform administrator to resolve permissions issues. 

Best,
Guillaume

**Reply by hyeom-wisc** [reply]
Guillaume, 

Thank you for the comment. I tried that and it shows as follows:

(moose) hyeom@ERB914-7920:~/projects/babbler/problems$ chmod +rw pressure_diffusion.i
(moose) hyeom@ERB914-7920:~/projects/babbler/problems$ ls -l
total 4
-rw-r--r-- 1 hyeom hyeom 1544 Nov 27 12:27 pressure_diffusion.i

Was it right command?

We have an workstation which group members share with through individual's college account. I am a first user to try to run MOOSE using the workstation. 

Sincerely,
Sung

**Comment by GiudGiud** [comment]
It looks right. Can you run the file now?

**Reply by hyeom-wisc** [reply]
Guillaume,
Thanks. I am not sure there is another issue now. I just copy and paste the code in ATOM and saved as "pressure_diffusion.i" shown in the tutorial. Please see below. Sung

```
(moose) hyeom@ERB914-7920:~/projects/babbler/problems$ ../babbler-opt -i pressure_diffusion.i

*** ERROR ***
/home/hyeom/projects/babbler/problems/pressure_diffusion.i contains Windows(DOS) line endings which are not supported.

Stack frames: 10
0: libMesh::print_trace(std::ostream&)
1: moose::internal::mooseErrorRaw(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)
2: void mooseError<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&&)
3: MooseUtils::checkFileReadable(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool, bool)
4: Parser::parse(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
5: MooseApp::setupOptions()
6: MooseApp::run()
7: ../babbler-opt(+0x41d1) [0x7f8fc56b01d1]
8: __libc_start_main
9: ../babbler-opt(+0x44af) [0x7f8fc56b04af]
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0
[unset]: write_line error; fd=-1 buf=:cmd=abort exitcode=1
:
system msg for write_line failure : Bad file descriptor
```

**Reply by dschwen** [reply]
run `dos2unix pressure_diffusion.i` if available

**Reply by hyeom-wisc** [reply]
Dear dschwen,
I typed dos2unix pressure_diffusion.i in the prompt window. It said Command 'dos2unix' not found... I tried run dos2unix pressure_diffusion.i. It also said Command 'run' not found. I think I messed up something during installations of ATOM and Github. Sung

**Reply by GiudGiud** [reply]
Hi Sung,

This is a common issue with Windows adding end-of-line characters to text files.
Please try the solutions listed here [se_link](https://unix.stackexchange.com/questions/32001/what-is-m-and-how-do-i-get-rid-of-it) as you don't have dos2unix installed on your system. 

Guillaume

**Reply by hyeom-wisc** [reply]
Thank you all.

I have installed "dos2unix" using the information shown in the link below (for future new users)
https://zoomadmin.com/HowToInstall/UbuntuPackage/dos2unix

I can run the input file after converting the file to unix format. Thanks. Sung

**Reply by CGH20171006** [reply]
> yep,i meet the same error. the dos2unix can fix the error, and i can also use vs code to change it(maybe this will be more convenient)
![image](https://user-images.githubusercontent.com/107262618/180232561-8e43629e-3bfe-4472-9165-4fcf51f1fac5.png)
